test/sshkit/scp_functional_test.exs

```diff
@@ -5,33 +5,167 @@ defmodule SSHKit.SCPFunctionalTest do
   alias SSHKit.SSH
 
   @defaults [silently_accept_hosts: true]
-  @local_workspace "test/fixtures/local_workspace"
-  @remote_workspace "/workspace"
-  @local_remote_workspace "test/fixtures/docker_workspace"
-
-  @tag boot: 1
-  test "uploads a file", %{hosts: [host]} do
-    options = [port: host.port, user: host.user, password: host.password]
-    local = "#{@local_workspace}/local_file.txt"
-    remote = "#{@remote_workspace}/#{host.id}.file"
-    shared = "#{@local_remote_workspace}/#{host.id}.file"
-
-    {:ok, conn} = SSH.connect(host.ip, Keyword.merge(@defaults, options))
-    assert :ok = SCP.upload(conn, local, remote)
-    assert File.read(local) == File.read(shared)
-    File.rm!(shared)
-  end
-
-  @tag boot: 1
-  test "downloads a file", %{hosts: [host]} do
-    options = [port: host.port, user: host.user, password: host.password]
-    remote = "#{@remote_workspace}/remote_file.txt"
-    local = "#{@local_workspace}/#{host.id}.file"
-    shared = "#{@local_remote_workspace}/remote_file.txt"
-
-    {:ok, conn} = SSH.connect(host.ip, Keyword.merge(@defaults, options))
-    assert :ok = SCP.download(conn, remote, local)
-    assert File.read(shared) == File.read(local)
-    File.rm!(local)
+
+  describe "upload/4" do
+    @tag boot: 1
+    test "sends a file", %{hosts: [host]} do
+      options = [port: host.port, user: host.user, password: host.password]
+      local = "test/fixtures/local_workspace/local_file.txt"
+      remote = "file.txt"
+
+      {:ok, conn} = SSH.connect(host.ip, Keyword.merge(@defaults, options))
+
+      assert :ok = SCP.upload(conn, local, remote)
+      assert verify_transfer(conn, local, remote)
```

Comment:
☝️ I've created [an issue](https://github.com/bitcrowd/sshkit.ex/issues/72) -- we can discuss there